lib/ResolverFactory.js

```diff
@@ -241,6 +246,7 @@ exports.createResolver = function(options) {
 	resolver.ensureHook("file");
 	resolver.ensureHook("finalFile");
 	resolver.ensureHook("existingFile");
+	resolver.ensureHook("finalResolved");
```

Comment:
I think you don't need a new hook here. You could place the restrictions plugin before the result plugin in the resolved hook.

In this case the restrictions plugin don't need a `target`, it would only error when needed and pass the control flow further (with `callback()`) when everything is fine.